### ISSUE TITLE
Need to generalise data input mechanisms.

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#375]
We currently use DATA_SOURCE which points to a directory containing subdirectories `reads`, `variants` and `references`. One possibility would be to have a `REFERENCES` config variable that is a list of directories (or SQLite DBs for graph mode?), each containing a single `ReferenceSet`. We could have another variable `DATASETS` which contains a list of directories, each of which has the required `variants` and `reads` subdirectories. 

We should also consider the issues of auth down the line, and how we associate permissions with datasets (or at lease, choose a mechanism which will allow for this).

**Comment by dcolligan** [comment]
Why can't we have DATA_SOURCE point to a directory of structure:
```
/root
  /dataset1
    /reads
    /variants
    /references
  /dataset2
 ...
```
The three things that a dataset is supposed to enforce (access control, billing, provenance) seem to be applicable to all three data types, correct?

**Comment by jeromekelleher** [comment]
This is what we started out with initially, but various people thought that it was important that each dataset is associated with it's own independent directory. I don't really have a strong opinion on this: I think we should play this by ear, and see what works out as the simplest approach when we start adding ACLs to the datasets.

One thing though is that `references` are *not* contained in a dataset. These are definitely a separate thing (we expect lots of datasets, and one or two referencesets, at most).

**Comment by dcolligan** [comment]
Ok, why don't we keep the single DATA_SOURCE variable and have it point to a directory tree of this structure:
```
/root
  /references
  /dataset1
    /reads
    /variants
  /dataset2
    /reads
    /variants
  ...
```
That is, any directory under the root not named "references" is considered a dataset.  This seems simpler to me than adding variables to our server config, and we're imposing a directory structure on the end user either way, so not go with the simpler one?

**Comment by jeromekelleher** [comment]
I'm happy with this, so let's go with your approach unless someone shouts out with a good reason not to.

**Comment by dcolligan** [comment]
Fixed in #420
